var/spack/repos/builtin/packages/totalview/package.py

```diff
@@ -3,184 +3,61 @@
+        "2024.2-powerle",
+        sha256="2bc1ef377e95f6f09d1f221a1dcc2f79415bad9e1e8403c647f38e2d383524d6",
+        url="https://dslwuu69twiif.cloudfront.net/totalview/2024.2/totalview_2024.2.11_linux_powerle.tar",
     )
 
-    # Version 2021.4.10
-    resource(
-        name="x86-64",
-        url="file://totalview.2021.4.10-linux-x86-64.tar",
-        destination=".",
-        sha256="7e5509b2cfb219100b0032304bdad7d422657c0736c386ba64bdb1bf11d10a1d",
-        when="@2021.4.10 platform=linux target=x86_64:",
-    )
-    resource(
-        name="ppcle",
-        url="file://totalview_2021.4.10_linux_powerle.tar",
-        destination=".",
-        sha256="79e812d1cd600172c5ea29c4aa6fb660d293300683419af36dd0e52cd7e15d56",
-        when="@2021.4.10 platform=linux target=ppc64le:",
-    )
-    resource(
-        name="aarch64",
-        url="file://totalview_2021.4.10_linux_arm64.tar",
-        destination=".",
-        sha256="46faaae1f33b4f4a20de345611092fbc65cd5759511c1fcf86ca71a0811c76fd",
-        when="@2021.4.10 platform=linux target=aarch64:",
-    )
-    resource(
-        name="darwinx86",
-        url="file://totalview.2021.4.10-darwin-x86.tar",
-        destination=".",
-        sha256="adbf95f86763e3cc5ec51fd504f3172bdcbb42a7f1f4e73b17cacca002729ad2",
-        when="@2021.4.10 platform=darwin target=x86_64:",
+    version(
+        "2024.2-linux-arm64",
+        sha256="63f737e61c2fb7f4816bcfc1d00e9e7c39817455531abdd09500f953be4ac75d",
+        url="https://dslwuu69twiif.cloudfront.net/totalview/2024.2/totalview_2024.2.11_linux_arm64.tar",
     )
 
-    # Version 2021.3
-    resource(
-        name="x86_64",
-        url="file://totalview_2021.3.9_linux_x86-64.tar",
-        destination=".",
-        sha256="6315ca855e1bee14678c640c3c9c8207b4f66c91714dcedd4aed592354112b48",
-        when="@2021.3.9 platform=linux target=x86_64:",
+    version(
+        "2024.1-x86-64",
+        sha256="964b73e70cb9046ce320bb0f95891b05c96a59117e5243fdc269855831c7059b",
+        url="https://dslwuu69twiif.cloudfront.net/totalview/2024.1/totalview_2024.1.21_linux_x86-64.tar",
     )
-    resource(
-        name="ppcle",
-        url="file://totalview_2021.3.9_linux_powerle.tar",
-        destination=".",
-        sha256="a7657b61895805024f5d4e7550796a485f98ce297a585583cdd20fe0b9b30bbb",
-        when="@2021.3.9 platform=linux target=ppc64le:",
+
+    version(
+        "2024.1-powerle",
+        sha256="c4dd8a3099d4f6ed23a6646b1d091129e0bf0b10c7a0d7ec73bd767818bab39b",
+        url="https://dslwuu69twiif.cloudfront.net/totalview/2024.1/totalview_2024.1.21_linux_powerle.tar",
     )
 
-    def url_for_version(self, version):
-        return "file://{0}/totalview.{1}-doc.tar".format(os.getcwd(), version)
+    version(
+        "2024.1-linux-arm64",
+        sha256="769527478dceb30855413970621f09a9dc54ef863ddaf75bb5a40142a54af346",
+        url="https://dslwuu69twiif.cloudfront.net/totalview/2024.1/totalview_2024.1.21_linux_arm64.tar",
+    )
 
```

Comment:
In spack we prefer not removing versions without deprecation in a release first since users value reproducibility in their environments. Please keep the previous versions (with local file or aws bucket download, if those are also available). Can you keep the older versions, maybe in the new structure?